### DISCUSSION TITLE
Functional component has an error with ref (register, Controller)

### DISCUSSION THREAD

**Discussion by ivanafanasyeu** [react-hook-form/react-hook-form#5869] [Q&A] [answered]
**Describe the bug**
Functional custom components has an error with refs. 
If say more correct, it's a bad functional components support from the box

**To Reproduce**

1. Create an **own** custom input as a functional component
2. Pass it to render inside Controller or use it with {...register}

**Codesandbox link (Required)**
https://codesandbox.io/s/fc-jqt1y?file=/src/App.tsx

**V7**

**Expected behavior**
No warnings in console when I use code examples

**Screenshots**
<img width="568" alt="Screenshot 2021-07-07 at 17 44 30" src="https://user-images.githubusercontent.com/25664253/124780077-fcfdd680-df4a-11eb-9766-355ba85f5096.png">

p.s. Easy and proper pass props, or an example in docs for own inputs. You suppose to know, that most people use functional components approach in coding and I think it would be better to have a great support for them, instead of additional not important code.

**Comment by bluebill1049** [comment, marked as answer]
This is a problem which external controlled component doesn't have a standard name for `ref` instead `inputRef`, `innerRef` and etc, there is not much we can have/control over that instead users may need to aware of that and assign correctly.
